src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -19,12 +17,45 @@
    :headers                   {"Content-Disposition" (format "attachment; filename=\"query_result_%s.xlsx\""
                                                              (u.date/format (t/zoned-date-time)))}})
 
+(defprotocol ExcelFormatValue
+  "Protocol for specifying how objects of various classes in QP result rows should be formatted for Excel
+  exports. The generic version in `common/FormatValue` does not work for Excel due to date/time formats."
+  (excel-format-value [this]
+    "Format this value in a QP result row appropriately for a results download to Excel."))
+
+;; this version of the protocol resolves #10803 by not converting datetimes to ISO8601 (which Excel cannot parse)
+;; Docjure can handle java.util.Date instances, but not the new Java 8 Time instances, so we convert everything
+;; to java.util.Date.
+(extend-protocol ExcelFormatValue
+  nil
+  (excel-format-value [_] nil)
+
+  Object
+  (excel-format-value [this] this)
+
+  java.time.temporal.Temporal
+  (excel-format-value [this]
+    (t/java-date this))
+
+  java.time.LocalDateTime
+  (excel-format-value [this]
+    ;; this is a sticky one - LocalDateTime where? Using the server's timezone for the conversion.
+    (t/java-date (t/zoned-date-time this (t/zone-id))))
+
+  java.time.OffsetDateTime
+  (excel-format-value [this]
+     (t/java-date this))
+
+  java.time.ZonedDateTime
+  (excel-format-value [this]
+    (t/java-date this)))
+
 ;; add a generic implementation for the method that writes values to XLSX cells that just piggybacks off the
 ;; implementations we've already defined for encoding things as JSON. These implementations live in
 ;; `metabase.middleware`.
 (defmethod spreadsheet/set-cell! Object [^Cell cell, value]
-  (when (= (.getCellType cell) Cell/CELL_TYPE_FORMULA)
-    (.setCellType cell Cell/CELL_TYPE_STRING))
+  (when (= (.getCellType cell) CellType/FORMULA)
+    (.setCellType cell CellType/STRING))
```

Comment:
These changes were to stop using the deprecated `CELL_TYPE_*` forms that are removed in POI 4.x. (would impact an upgrade)